frontend/src/main/web/src/components/OverlayTrigger/index.jsx

```diff
@@ -0,0 +1,294 @@
+import React, { cloneElement, Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import contains from 'dom-helpers/query/contains'
+import warning from 'warning'
+import { pick } from 'lodash'
+import createChainedFunction from '../../utils/createChainedFunction'
+import Overlay from '../Overlay'
+/**
+ * Check if value one is inside or equal to the of value
+ *
+ * @param {string} one
+ * @param {string|array} of
+ * @returns {boolean}
+ */
+function isOneOf (one, of) {
+  if (Array.isArray(of)) {
+    return of.indexOf(one) >= 0
+  }
+  return one === of
+}
+
+class OverlayTrigger extends Component {
+  state = {
+    isOverlayShown: this.props.defaultOverlayShown
+  }
+  show = () => {
+    this.setState({
+      isOverlayShown: true
+    })
+  }
+  hide = () => {
+    this.setState({
+      isOverlayShown: false
+    })
+  }
+  toggle = () => {
+    if (this.state.isOverlayShown) {
+      this.hide()
+    } else {
+      this.show()
+    }
+  }
+  renderOverlay = () => {
+    ReactDOM.unstable_renderSubtreeIntoContainer(
```

Comment:
This is not a normal thing to do. Please add a comment explaining why this is being used. Is `unstable_renderSubtreeIntoContainer` documented somewhere?